src/generators.rs

```diff
@@ -1072,6 +1072,393 @@ pub fn directed_binomial_tree_graph(
+///         |   |   |
+///     ... F-S-F   F-...
+///         |   |   |
+///     ... D   D-S-D ...
+///
+///
+/// :param int d: distance of the code.
+/// :param bool multigraph: When set to False the output
+///     :class:`~retworkx.PyGraph` object will not be not be a multigraph and
+///     won't  allow parallel edges to be added. Instead
+///     calls which would create a parallel edge will update the existing edge.
+///
+/// :returns: The generated heavy hex directed graph
+/// :rtype: PyDiGraph
+/// :raises IndexError: If d is even.
+///
+/// .. jupyter-execute::
+///
+///   import os
+///   import tempfile
+///
+///   import pydot
+///   from PIL import Image
+///
+///   import retworkx.generators
+///
+///   graph = retworkx.generators.heavy_hex_graph(3)
+///   dot_str = graph.to_dot(
+///       lambda node: dict(
+///           color='black', fillcolor='lightblue', style='filled'))
+///   dot = pydot.graph_from_dot_data(dot_str)[0]
+///
+///   with tempfile.TemporaryDirectory() as tmpdirname:
+///       tmp_path = os.path.join(tmpdirname, 'dag.png')
+///       dot.write_png(tmp_path)
+///       image = Image.open(tmp_path)
+///       os.remove(tmp_path)
+///   image
+///
+#[pyfunction(bidirectional = false, multigraph = true)]
+#[pyo3(text_signature = "(d, /, bidirecational=False, multigraph=True)")]
+pub fn directed_heavy_hex_graph(
+    py: Python,
+    d: usize,
+    bidirectional: bool,
+    multigraph: bool,
+) -> PyResult<digraph::PyDiGraph> {
+    let mut graph = StableDiGraph::<PyObject, PyObject>::default();
+
+    if d % 2 == 0 {
+        return Err(PyIndexError::new_err("d must be odd"));
+    }
+
+    let num_data = d * d;
+    let num_syndrome = (d - 1) * (d + 1) / 2;
+    let num_flag = d * (d - 1);
+
+    let nodes_data: Vec<NodeIndex> =
+        (0..num_data).map(|_| graph.add_node(py.None())).collect();
+    let nodes_syndrome: Vec<NodeIndex> = (0..num_syndrome)
+        .map(|_| graph.add_node(py.None()))
+        .collect();
+    let nodes_flag: Vec<NodeIndex> =
+        (0..num_flag).map(|_| graph.add_node(py.None())).collect();
+
+    // connect data and flags
+    for (i, flag_chunk) in nodes_flag.chunks(d - 1).enumerate() {
+        for (j, flag) in flag_chunk.iter().enumerate() {
+            graph.add_edge(nodes_data[i * d + j], *flag, py.None());
+            graph.add_edge(*flag, nodes_data[i * d + j + 1], py.None());
+            if bidirectional {
+                graph.add_edge(*flag, nodes_data[i * d + j], py.None());
+                graph.add_edge(nodes_data[i * d + j + 1], *flag, py.None());
+            }
+        }
+    }
+
+    // connect data and syndromes
+    for (i, syndrome_chunk) in nodes_syndrome.chunks((d + 1) / 2).enumerate() {
+        if i % 2 == 0 {
+            graph.add_edge(nodes_data[i * d], syndrome_chunk[0], py.None());
+            graph.add_edge(
+                syndrome_chunk[0],
+                nodes_data[(i + 1) * d],
+                py.None(),
+            );
```

Comment:
```suggestion
            graph.add_edge(
                nodes_data[(i + 1) * d],
                syndrome_chunk[0],
                py.None(),
            );
```

Reply:
I'm like ~95% confident that this suggested change is right. The circuit described in the paper would prefer the other natural direction on this small set of edges, but if natural source/targets are required to alternate in hardware, then having to simulate flipping these edges through Hadamards is preferable to flipping the alternating pattern through the bulk.

But there is a 5% chance that I'm missing something.